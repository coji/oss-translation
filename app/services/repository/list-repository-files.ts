import fg from 'fast-glob'
import { getRepositoryFileContent } from './get-repository-file-content'
import type { RepositoryFile } from './types'

export const listRepositoryFiles = async (directory: string) => {
  const filenames = await fg('**/*', {
    cwd: directory,
    onlyFiles: true,
  })

  const files: RepositoryFile[] = []

  for (const filename of filenames) {
    const result = await getRepositoryFileContent(directory, filename)
    if (result.isOk()) {
      files.push(result.value)
    }
  }

  return files
}
