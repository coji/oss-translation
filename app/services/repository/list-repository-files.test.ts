import path from 'node:path'
import { expect, test } from 'vitest'
import { listRepositoryFiles } from './list-repository-files'

test('listRepositoryFiles', async () => {
  const directory = path.join(
    import.meta.dirname,
    'test/fixtures/projects/projectA/docs',
  )
  const files = await listRepositoryFiles(directory)
  expect(files).toMatchSnapshot()
})
