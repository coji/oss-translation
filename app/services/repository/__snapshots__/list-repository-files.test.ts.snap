// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`listRepositoryFiles 1`] = `[]`;
